src/Services/AppInstallActivationHandler.cs

```diff
@@ -51,63 +63,107 @@ protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
-            if (_setupFlowOrchestrator.IsMachineConfigurationInProgress)
+            var searchQuery = parameters.Get("search");
+            var wingetIdsQuery = parameters.Get("wingetIds");
+
+            if (!string.IsNullOrEmpty(searchQuery))
             {
-                _log.Warning("Cannot activate the add-apps-to-cart flow because the machine configuration is in progress");
-                return;
+                await AppActivationFlowAsync(searchQuery, ActivationQueryType.Search);
             }
-            else
+            else if (!string.IsNullOrEmpty(wingetIdsQuery))
             {
-                _log.Information("Starting add-apps-to-cart activation");
-                _navigationService.NavigateTo(typeof(SetupFlowViewModel).FullName!);
-                _setupFlowViewModel.StartAppManagementFlow(query);
-                await SearchAndSelectAsync(query);
+                await AppActivationFlowAsync(wingetIdsQuery, ActivationQueryType.WingetIds);
             }
         }
-        catch (Exception ex)
-        {
-            _log.Error(ex, "Error executing the add-apps-to-cart activation flow");
-        }
     }
 
-    private async Task SearchAndSelectAsync(string query)
+    private async Task AppActivationFlowAsync(string query, ActivationQueryType queryType)
     {
-        var parameters = HttpUtility.ParseQueryString(query);
-        var searchParameter = parameters["search"];
+        if (_setupFlowOrchestrator.IsMachineConfigurationInProgress)
+        {
+            _log.Warning("Cannot activate the add-apps-to-cart flow because the machine configuration is in progress");
+            await _mainWindow.ShowErrorMessageDialogAsync(
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.AppInstallActivationTitle),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.URIActivationFailedBusy),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.Close));
+            return;
+        }
 
-        if (string.IsNullOrEmpty(searchParameter))
+        var identifiers = ParseIdentifiers(query, queryType);
+        if (identifiers.Length == 0)
         {
-            _log.Warning("Search parameter is missing or empty in the query.");
+            _log.Warning("No valid identifiers provided in the query.");
             return;
         }
 
-        // Currently using the first search term only
-        var firstSearchTerm = searchParameter.Split(Separator, StringSplitOptions.RemoveEmptyEntries)
-                                             .Select(term => term.Trim(' ', '"'))
-                                             .FirstOrDefault();
+        _log.Information("Starting add-apps-to-cart activation");
+        _navigationService.NavigateTo(typeof(SetupFlowViewModel).FullName!);
+        _setupFlowViewModel.StartAppManagementFlow(queryType == ActivationQueryType.Search ? identifiers[0] : null);
```

Comment:
why null if `ActivationQueryType` isn't search?

Reply:
That's because we only want to do the query prefilling into the search box piece in the search scenario.

Reply:
similar to the other comment, lmk if you think using a bool `prefillSearch` might help with clarity here.

Reply:
Do you know if null is passed anywhere else into `StartAppManagementFlow`?  [MainPageViewModel.StartAppManagementFlow](https://github.com/microsoft/devhome/blob/main/tools/SetupFlow/DevHome.SetupFlow/ViewModels/MainPageViewModel.cs#L108) throws if the string is null when trying to log the query.

Reply:
this is the only place we call StartAppManagementFlow -- it was added with the previous PR: https://github.com/microsoft/devhome/pull/2642/files

With the change to not log the query string, StartAppManagementFlow should not throw if null (tested locally to confirm)